.travis.yml

```diff
@@ -51,11 +51,10 @@ matrix:
     # has (on-screen) OpenGL installed, we need to setup environment variable
     # to avoid having the linker load the wrong libglapi.so which would cause
     # OSMesa to crash
-    - env: PYTHON=2.7 DEPS=full TEST=osmesa
+    - env: PYTHON=3.6 DEPS=osmesa TEST=osmesa
```

Comment:
Do the osmesa-based tests not need the "full" dependencies?

Reply:
@djhoese I removed the system (apt) libs to not give chance for clashes, because I use `mesalib` and `libglu` from conda-forge.

Reply:
I see you removed `full_apt` (can the whole `addons` section be removed?), but changing `DEPS=full` to `DEPS=osmesa` means that pyqt, matplotlib, imageio, and other dependencies won't be installed, right? Is that expected?

Reply:
These are not needed for osmesa, I think.

Reply:
No, the whole section can't be removed. I did it, then the main apt-section takes over.

Reply:
Ok I just want to make sure that tests aren't being skipped due to missing dependencies so they look like they are passing. Similar thing happened on the PyQt5 PR where the tests were still running with PyQt4... 😭

Reply:
Yes, this need to be checked, but I'm quite confident, that the main dependencies are installed with the first conda install batch. Will check anyway.

Reply:
I would suggest, to check the dependencies for each backend and create the conda-forge env accordingly in travis. Do you think that can be postponed to another PR? I checked the builds from 4 weeks ago, they seem all the same when it comes to collected, deselected and selected.

Reply:
I checked the total number of tests run for a previous success of the osmesa environment and for this new one and they have the same number passed, skipped, and deselected. So I'm good to go. Do you want to fix that if statement on line 144 or should be just merge this?

Reply:
Let me just push the fix for line 144.